app/(routes)/pubhub/v1/user/loans/[identifier]/route.ts

```diff
@@ -11,20 +11,28 @@ async function createdigitalLoan(
   request: NextRequest,
   context: { uniLoginUserInfo: TUserInfo; params: { identifier: string } }
 ) {
-  const { identifier } = context.params
+  const { identifier } = await context.params
   const createLoan = createLoanSchema.transform(loanData => {
     const response = loanData.response
     return {
-      expirationDateUtc: transformTimeToUtcString(response.data.expirationdateutc),
-      orderNumber: response.data.internalordernumber,
-      orderId: response.data.retailerordernumber,
+      expirationDateUtc: transformTimeToUtcString(response.data?.expirationdateutc || ""),
+      orderNumber: response.data?.internalordernumber,
+      orderId: response.data?.retailerordernumber,
       code: response.status.code,
       message: `${response.status.message} (#${response.status.code})`,
     }
   })
   try {
     const responseData = await createLoanRequest(context.uniLoginUserInfo, identifier)
-    return NextResponse.json(createLoan.parse(responseData))
+    const data = createLoan.parse(responseData)
+
+    // Check if the response code is not "101" (OK)
+    // https://docs.pubhub.dk/LibraryApi/2.7/Pubhub_Library_Integration_Guide_v2.7_September_2023.pdf
+    if (data.code !== "101") {
```

Comment:
Nice with the comment. I think it would make sense to use the response code from the [generated code](https://github.com/danskernesdigitalebibliotek/dpl-go/blob/5b777a0f5a7b0ef554c54cc503f9ba35dfb54d01/lib/rest/publizon/local-adapter/generated/model/apiResponseCode.ts#L17) to make it clear what it is.